tests/lib/subgraph/DeploySubgraph.test.ts

```diff
@@ -0,0 +1,339 @@
+      auxiliaryChain: '1405',
+      subgraphType: SubGraphType.AUXILIARY,
+      graphAdminRPC: 'http://localhost:8020',
+      graphIPFS: 'http://localhost:5001',
+      mosaicConfigPath: '~/.mosaic/ropsten/mosaic.json',
+      gatewayAddress: '0x0000000000000000000000000000000000000001',
+      gatewayConfigPath: '~/.mosaic/ropsten/1405/0x0000000000000000000000000000000000000001,json',
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should fail if mosaic config not found at default directory', () => {
+    input.mosaicConfigPath = undefined;
+    input.gatewayAddress = undefined;
+    input.gatewayConfigPath = undefined;
+    sinon.replace(
+      MosaicConfig,
+      'exists',
+      sinon.fake.returns(false),
+    );
+
+    assert.throws(
+      () => deploySubGraph(
+        input.originChain,
+        input.auxiliaryChain,
+        input.subgraphType,
+        input.graphAdminRPC,
+        input.graphIPFS,
+        input.mosaicConfigPath,
+        input.gatewayAddress,
+        input.gatewayConfigPath,
+      ),
+      'Mosaic config or gateway config not found . Use --mosaic-config or --gateway-config option to provide path.',
+    );
+  });
+
+  it('should fail if mosaic config origin chain is different from passed value', () => {
+    input.mosaicConfigPath = undefined;
+    input.gatewayAddress = undefined;
+    input.gatewayConfigPath = undefined;
+    sinon.replace(
+      MosaicConfig,
+      'exists',
+      sinon.fake.returns(true),
+    );
+
+    const fakeMosaicConfig = { originChain: { chain: 'goerli' } };
+    sinon.replace(
+      MosaicConfig,
+      'fromChain',
+      sinon.fake.returns(fakeMosaicConfig),
+    );
+
+    assert.throws(
+      () => deploySubGraph(
+        input.originChain,
+        input.auxiliaryChain,
+        input.subgraphType,
+        input.graphAdminRPC,
+        input.graphIPFS,
+        input.mosaicConfigPath,
+        input.gatewayAddress,
+        input.gatewayConfigPath,
+      ),
+      `Origin chain id in mosaic config is ${fakeMosaicConfig.originChain.chain} but received argument is ${input.originChain}`,
+    );
+  });
+
+  it('should fail if auxiliary chain id in gateway config doesnot match passed value', () => {
+    input.mosaicConfigPath = undefined;
+    input.gatewayAddress = undefined;
+    input.gatewayConfigPath = path.join(
+      Directory.projectRoot,
+      'tests/Config/testdata/0xae02c7b1c324a8d94a564bc8d713df89eae441fe.json',
+    );
+    assert.throws(
+      () => deploySubGraph(
+        input.originChain,
+        input.auxiliaryChain,
+        input.subgraphType,
+        input.graphAdminRPC,
+        input.graphIPFS,
+        input.mosaicConfigPath,
+        input.gatewayAddress,
+        input.gatewayConfigPath,
+      ),
+      `Auxiliary chain id in gateway config is 1000 but value passed is ${input.auxiliaryChain}`,
+    );
+  });
+
+  it('should fail if mosaic config doesnot exist on mentioned path', () => {
```

Comment:
```suggestion
  it('should fail if mosaic config does not exist on mentioned path', () => {
```
Please see other statements also.